.github/workflows/vulncheck.yaml

```diff
@@ -14,21 +14,23 @@ on:
 
 jobs:
   govulncheck:
+    if: github.base_ref != 'main'
     name: detect
     runs-on: ubuntu-latest
     env:
       GOPATH: ${{ github.workspace }}
     steps:
-      - name: Set up Go 1.21.x
-        uses: actions/setup-go@v4
-        with:
-          go-version: 1.21.x
-
       - name: Checkout
         uses: actions/checkout@v2
         with:
           path: ./src/github.com/${{ github.repository }}
 
+      - name: Setup Golang
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: ./src/github.com/${{ github.repository }}/go.mod
```

Comment:
This is ok as long as upstream we keep everything up to date. Otherwise if upstream(midstream) we deviate from the build version (go mod version is outdated) then this check wil be outdated too. At the end of the day for this action only the version that you build against matters and also you need to look ahead a bit so you know what vulnerabilities are coming, which we might miss.